ql/src/java/org/apache/hadoop/hive/ql/ddl/view/materialized/drop/DropMaterializedViewAnalyzer.java

```diff
@@ -50,7 +54,14 @@ public void analyzeInternal(ASTNode root) throws SemanticException {
     Table materializedView = getTable(viewName, throwException);
     if (materializedView != null) {
       inputs.add(new ReadEntity(materializedView));
-      outputs.add(new WriteEntity(materializedView, WriteEntity.WriteType.DDL_EXCLUSIVE));
+
+      boolean tableWithSuffix = (HiveConf.getBoolVar(conf, ConfVars.HIVE_ACID_CREATE_TABLE_USE_SUFFIX)
+          || HiveConf.getBoolVar(conf, ConfVars.HIVE_ACID_LOCKLESS_READS_ENABLED))
+        && AcidUtils.isTransactionalTable(materializedView)
+        && Boolean.parseBoolean(materializedView.getProperty(SOFT_DELETE_TABLE));
+      
+      outputs.add(new WriteEntity(materializedView,
+        tableWithSuffix ? WriteType.DDL_EXCL_WRITE : WriteType.DDL_EXCLUSIVE));
```

Comment:
I found that this piece of code is also present in `DropTableAnalyzer`
https://github.com/apache/hive/blob/cba74fa91d4035aee2c39dcf929a0ae480609540/ql/src/java/org/apache/hadoop/hive/ql/ddl/table/drop/DropTableAnalyzer.java#L59

Could you please try reusing it? Like extracting it into some utility class or into the common super class  `BaseSemanticAnalyzer`

Reply:
refactored